extensions/opentelemetry/opentelemetry-async-mutiny/runtime/src/main/java/io/quarkus/opentelemetry/async/mutiny/runtime/tracing/TracingMultiSubscriber.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.opentelemetry.async.mutiny.runtime.tracing;
+
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.smallrye.mutiny.subscription.MultiSubscriber;
+
+public class TracingMultiSubscriber<T> implements MultiSubscriber<T> {
+
+    private final Context context;
+
+    private final Subscriber<? super T> subscriber;
+
+    public TracingMultiSubscriber(final Context context, final Subscriber<? super T> subscriber) {
+        this.context = context;
+        this.subscriber = subscriber;
+    }
+
+    @Override
+    public void onSubscribe(final Subscription subscription) {
+        subscriber.onSubscribe(subscription);
+    }
+
+    @Override
+    public void onItem(final T item) {
+        try (final Scope ignored = context.makeCurrent()) {
+            subscriber.onNext(item);
+        }
```

Comment:
Is it expected to close the scope? onNext is likely going to do async operation. So, the return of onNext does not mean we are done.

Reply:
Yes this way necessary. See also: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/2b6e89b6eb6f6671a19430bcb219e5ef65f5dfc4/instrumentation/rxjava/rxjava-2.0/library/src/main/java/io/opentelemetry/instrumentation/rxjava/v2_0/TracingObserver.java#L33